GPU/GLES/GLES_GPU.cpp

```diff
@@ -1915,15 +1928,18 @@ void GLES_GPU::Execute_Generic(u32 op, u32 diff) {
 }
 
 void GLES_GPU::FastLoadBoneMatrix(u32 target) {
+	const int num = gstate.boneMatrixNumber & 0x7F;
+	const int mtxNum = num / 12;
+	uint32_t uniformsToDirty = DIRTY_BONEMATRIX0 << mtxNum;
+	if ((num - 12 * mtxNum) != 0) {
+		uniformsToDirty |= DIRTY_BONEMATRIX0 << ((mtxNum + 1) & 7);
+	}
+
 	if (!g_Config.bSoftwareSkinning || (gstate.vertType & GE_VTYPE_MORPHCOUNT_MASK) != 0) {
 		Flush();
-		const int num = gstate.boneMatrixNumber & 0x7F;
-		int mtxNum = num / 12;
-		uint32_t uniformsToDirty = DIRTY_BONEMATRIX0 << mtxNum;
-		if ((num - 12 * mtxNum) != 0) {
-			uniformsToDirty |= DIRTY_BONEMATRIX0 << ((mtxNum + 1) & 7);
-		}
 		shaderManager_->DirtyUniform(uniformsToDirty);
+	} else {
+		gstate_c.deferredVertTypeDirty |= uniformsToDirty;
```

Comment:
Technically we could avoid computing uniformsToDirty at all in the case of softskinning and no morph, but not sure if it's worth the extra check.